frontend/src/components/ui/textarea.tsx

```diff
@@ -34,4 +35,58 @@ const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
 );
 Textarea.displayName = "Textarea";
 
+export const DebouncedTextarea = React.forwardRef<
+  HTMLTextAreaElement,
+  TextareaProps & {
+    value: string;
+    onValueChange: (value: string) => void;
+    delay?: number;
+  }
+>(({ className, onValueChange, ...props }, ref) => {
+  const { value, onChange } = useDebounceControlledState<string>({
+    initialValue: props.value,
+    delay: props.delay,
+    onChange: onValueChange,
+  });
+
+  return (
+    <Textarea
+      ref={ref}
+      className={className}
+      {...props}
+      onChange={(evt) => onChange(evt.target.value)}
+      value={value}
+    />
+  );
+});
+DebouncedTextarea.displayName = "DebouncedTextarea";
+
+export const OnBlurredTextarea = React.forwardRef<
+  HTMLTextAreaElement,
+  TextareaProps & {
+    value: string;
+    onValueChange: (value: string) => void;
+  }
+>(({ className, onValueChange, ...props }, ref) => {
+  const [internalValue, setInternalValue] = React.useState(props.value);
```

Comment:
similar, might need `useControllableState` in case the value is changed externally from another input

Reply:
i think if you render the text input twice, this bug might appear (i might be wrong)

Reply:
you're right, it does -- let me go ahead and fix that

Reply:
okay, fixed the issue in my latest commit but maybe I did more than I needed to... let me know